org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/JavadocTagsSubProcessor.java

```diff
@@ -14,440 +14,139 @@
-import org.eclipse.jdt.core.dom.Javadoc;
-import org.eclipse.jdt.core.dom.MethodDeclaration;
-import org.eclipse.jdt.core.dom.Name;
-import org.eclipse.jdt.core.dom.SingleVariableDeclaration;
-import org.eclipse.jdt.core.dom.StructuralPropertyDescriptor;
-import org.eclipse.jdt.core.dom.TagElement;
-import org.eclipse.jdt.core.dom.TextElement;
-import org.eclipse.jdt.core.dom.Type;
-import org.eclipse.jdt.core.dom.TypeDeclaration;
-import org.eclipse.jdt.core.dom.TypeParameter;
-import org.eclipse.jdt.core.dom.VariableDeclaration;
-import org.eclipse.jdt.core.dom.VariableDeclarationFragment;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.ModuleDeclaration;
 import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
-import org.eclipse.jdt.core.dom.rewrite.ListRewrite;
-import org.eclipse.jdt.core.manipulation.CodeGeneration;
-import org.eclipse.jdt.internal.core.manipulation.dom.ASTResolving;
-import org.eclipse.jdt.internal.corext.dom.ASTNodes;
-import org.eclipse.jdt.internal.corext.dom.Bindings;
 import org.eclipse.jdt.internal.ui.text.correction.AddAllMissingJavadocTagsProposalCore;
 import org.eclipse.jdt.internal.ui.text.correction.AddJavadocCommentProposalCore;
 import org.eclipse.jdt.internal.ui.text.correction.AddMissingJavadocTagProposalCore;
+import org.eclipse.jdt.internal.ui.text.correction.AddMissingModuleJavadocTagProposalCore;
 import org.eclipse.jdt.internal.ui.text.correction.IInvocationContextCore;
 import org.eclipse.jdt.internal.ui.text.correction.IProblemLocationCore;
-import org.eclipse.jdt.internal.ui.text.correction.IProposalRelevance;
-import org.eclipse.jdt.ls.core.internal.Messages;
-import org.eclipse.jdt.ls.core.internal.corrections.CorrectionMessages;
+import org.eclipse.jdt.internal.ui.text.correction.JavadocTagsBaseSubProcessor;
+import org.eclipse.jdt.internal.ui.text.correction.ProblemLocation;
+import org.eclipse.jdt.internal.ui.text.correction.proposals.ReplaceCorrectionProposalCore;
 import org.eclipse.jdt.ls.core.internal.corrections.ProposalKindWrapper;
 import org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler;
 import org.eclipse.jdt.ui.text.java.correction.ASTRewriteCorrectionProposalCore;
 import org.eclipse.lsp4j.CodeActionKind;
-import org.eclipse.text.edits.TextEditGroup;
 
 
 /**
  *
  */
-public class JavadocTagsSubProcessor {
+public class JavadocTagsSubProcessor extends JavadocTagsBaseSubProcessor<ProposalKindWrapper> {
 	public static void getMissingJavadocTagProposals(IInvocationContextCore context, IProblemLocationCore problem, Collection<ProposalKindWrapper> proposals) {
-		ASTNode node= problem.getCoveringNode(context.getASTRoot());
-		if (node == null) {
-			return;
-		}
-		node= ASTNodes.getNormalizedNode(node);
-
-		BodyDeclaration bodyDeclaration= ASTResolving.findParentBodyDeclaration(node);
-		if (bodyDeclaration == null) {
-			return;
-		}
-		Javadoc javadoc= bodyDeclaration.getJavadoc();
-		if (javadoc == null) {
-			return;
-		}
```

Comment:
In the previous behaviour, this ensured we never suggested adding new javadoc where one already existed. With the new behaviour we always suggest it. Seems wrong.

It looks like this is missing upstream.